<launch>
    <arg name="odom_topic" default="/volta_1/odom"/>
    <arg name="gmapping" default="false"/>
    <arg name="scan_topic" default="/volta_1/scan_filtered" />
    <arg name="base_global" default="global_planner/GlobalPlanner"/>
    <arg name="base_local" default="dwa_local_planner/DWAPlannerROS"/>

    <arg name="amcl_use_map_topic" default="true"/>
    <arg name="odom_frame" default="volta_1_tf/odom"/>
    <arg name="base_frame" default="volta_1_tf/base_link"/>
    <arg name="map_frame" default="map"/>
    <arg name="amcl_tf_broadcast" default="true"/>
    <arg name="amcl_init_pose_x" default="0"/>
    <arg name="amcl_init_pose_y" default="0"/>
    <arg name="amcl_init_pose_yaw" default="0"/>
    <!-- <arg name="map_file" default="$(find multirobot)/maps/office/office.yaml"/> -->

    <node pkg="amcl" type="amcl" name="amcl1" unless="$(arg gmapping)">
        <rosparam file="$(find multirobot)/config/volta1/amcl.yaml" command="load"/>
        <param name="use_map_topic" value="$(arg amcl_use_map_topic)"/>
        <param name="global_frame_id" value="$(arg map_frame)"/>
        <param name="odom_frame_id" value="$(arg odom_frame)"/>
        <param name="base_frame_id" value="$(arg base_frame)"/>
        <param name="initial_pose_x" value="$(arg amcl_init_pose_x)"/>
        <param name="initial_pose_y" value="$(arg amcl_init_pose_y)"/>
        <param name="initial_pose_a" value="$(arg amcl_init_pose_yaw)"/>
        <param name="tf_broadcast" value="$(arg amcl_tf_broadcast)"/>
        <remap from="scan" to="$(arg scan_topic)"/>
    </node>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base1" output="screen">
        <remap from="odom" to="$(arg odom_topic)" />
        <remap from="cmd_vel" to="volta_1/cmd_vel" />

        <param name="base_global_planner" value="$(arg base_global)"/>
        <param name="base_local_planner" value="$(arg base_local)"/>

        <rosparam file="$(find multirobot)/config/volta1/global_planner.yaml" command="load"/>
        <rosparam file="$(find multirobot)/config/volta1/local_planner.yaml" command="load"/>

        <rosparam file="$(find multirobot)/config/volta1/costmap.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find multirobot)/config/volta1/costmap.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find multirobot)/config/volta1/costmap_local.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find multirobot)/config/volta1/costmap_global.yaml" command="load" ns="global_costmap"/>
    </node>
</launch>
